### ISSUE TITLE
 crash on import directory

### ISSUE THREAD

**Issue by plum** [olivierkes/manuskript#500]
As requested , here is a crash log from console, obatained upon a manuskript crash when attempting to import a directory. I am running Version 0.8.0 on archlinux (latest version).

The directory to imp[ort contains  28  .md files only,  in text format.
There are options to 'import under', and I chose 'top level directory', as well as no selection on different attempts. The result was the same, I.E.:

```
Debug: Web rendering engine used: QWebView
Running manuskript version 0.8.0.

Note: No translator found or loaded from system locale for locale en_GB.
Loading: /home/chuee/manuskript.msk
Detected file format version: 1. Zip: True.
Project /home/chuee/manuskript.msk loaded.

Traceback (most recent call last):
  File "/home/chuee/manuskript/bin/../manuskript/ui/importers/importer.py", line 134, in selectFile
    self.preview()
  File "/home/chuee/manuskript/bin/../manuskript/ui/importers/importer.py", line 240, in preview
    result = self.startImport(previewModel)
  File "/home/chuee/manuskript/bin/../manuskript/ui/importers/importer.py", line 298, in startImport
    self.settingsWidget)
  File "/home/chuee/manuskript/bin/../manuskript/importer/folderImporter.py", line 26, in startImport
    ext = self.getSetting("ext").value()
AttributeError: 'NoneType' object has no attribute 'value'
Fatal Python error: Aborted

Current thread 0x00007f148e4c4600 (most recent call first):
  File "/home/chuee/manuskript/bin/../manuskript/main.py", line 101 in launch
  File "/home/chuee/manuskript/bin/../manuskript/main.py", line 114 in run
  File "./bin/manuskript", line 13 in <module>
Aborted (core dumped)

```

**Comment by gedakc** [comment]
What **Format** did you choose?
(For example, Internal Markdown, Pandoc DocX)

So far I have been unable to reproduce the crash that you encountered.

I tested importing a markdown file using the "Import in a top-level directory" option and **Internal Markdown** format and it worked for me.

![manuskript-import-internal-markdown](https://user-images.githubusercontent.com/10405019/52902880-4e6b6680-31d4-11e9-9107-30cc1762bac1.png)

Perhaps you can provide a sample file that causes the crash on import?

**Comment by plum** [comment]
Ah, you misunderstand me. I am successful with attempt to import a SINGLE file in markdown format. Ther problem occured with attempt to import a DIRECTORY containing markdown files; The import directory option as obtained in the Format dropdown menu. Thei sthen obtains a browse button to select the directory for import, and so on.

**Comment by gedakc** [comment]
Thank you @plum for the clarification.  I was able to reproduce the crash with Manuskript 0.8.0 using folders and markdown files in the following structure:

```
-- MDFiles4Import --+-- Folder A --+-- Text-1.md
                    |              |
                    |              +-- Text-2.md
                    |
                    +-- Folder B --+-- Text-A.md
                                   |
                                   +-- Text B.md
```

Steps to reproduce crash:

1. Start manuskript and create a new project named "`test-import`" using the **Empty fiction** template.

2.  Select the **Editor** pane.

3. Choose **File -> Import**.

4. Choose options:  **Format** = _Folder_, and enable **Import in a top-level folder**.

5. Click on **Chose file** (this typo is fixed in the develop branch) and choose the "`MDFiles4Import`" directory.

![manuskript-import-folder-options](https://user-images.githubusercontent.com/10405019/52917123-f3ee0b00-32a4-11e9-8296-cedb3a97d937.png)

6. Note that Manusrkipt crashes with the following console log:

```
$ Project test-import.msk saved.
Project test-import.msk saved.
Traceback (most recent call last):
  File "/home/gedakc/workspace/manuskript.gedakc/bin/../manuskript/ui/importers/importer.py", line 134, in selectFile
    self.preview()
  File "/home/gedakc/workspace/manuskript.gedakc/bin/../manuskript/ui/importers/importer.py", line 240, in preview
    result = self.startImport(previewModel)
  File "/home/gedakc/workspace/manuskript.gedakc/bin/../manuskript/ui/importers/importer.py", line 298, in startImport
    self.settingsWidget)
  File "/home/gedakc/workspace/manuskript.gedakc/bin/../manuskript/importer/folderImporter.py", line 26, in startImport
    ext = self.getSetting("ext").value()
AttributeError: 'NoneType' object has no attribute 'value'
```

Labeling this issue as a bug.

**Comment by gedakc** [comment]
Using the steps listed above I have confirmed that this issue is fixed with the merge into the **develop** branch of PR #612 

Screen shot of Folder Import Preview:
![Manuskript-Import-Folder-Preview](https://user-images.githubusercontent.com/10405019/64480841-cdce8080-d18c-11e9-872d-4663177db223.png)

Screen shot after Folder is Imported:
![Manuskript-Import-Folder-Completed](https://user-images.githubusercontent.com/10405019/64480842-d9ba4280-d18c-11e9-8690-24748457eb20.png)

This improvement will be available in the upcoming 0.10.0 release.
